### ISSUE TITLE
Event Hubs processor fails to get events after a period of time

### ISSUE THREAD

**Issue by conniey** [Azure/azure-sdk-for-java#13420]
There are times when fetching events from Event Hubs using EventProcessor, events are not fetched and instead, a timeout exception is thrown. Consequently, the processor stops fetching events. In the repro, the event hub has 32 partitions.

```
INFO	2020-07-20 22:00:40,690	1806056	c.a.c.a.i.handler.SendLinkHandler	[single-1]	onLinkRemoteClose connectionId[MF_4ee337_1595280634612], linkName[eh0-int-stream], errorCondition[amqp:link:detach-forced], errorDescription[Idle link tracker, link eh0-int-stream has been idle for 1800000ms TrackingId:4d82b29d-b922-4634-a2ba-29fd14c40764_G16, SystemTracker:client-link18456132, Timestamp:2020-07-20T22:00:39]
INFO	2020-07-20 22:00:40,690	1806056	c.a.c.a.i.handler.SendLinkHandler	[single-1]	processOnClose connectionId[MF_4ee337_1595280634612], linkName[eh0-int-stream], errorCondition[amqp:link:detach-forced], errorDescription[Idle link tracker, link eh0-int-stream has been idle for 1800000ms TrackingId:4d82b29d-b922-4634-a2ba-29fd14c40764_G16, SystemTracker:client-link18456132, Timestamp:2020-07-20T22:00:39]
ERROR	2020-07-20 22:00:40,698	1806064	c.a.c.a.i.ReactorSender	[single-1]	[eh0-int-stream] Error occurred in sender error handler.
Idle link tracker, link eh0-int-stream has been idle for 1800000ms TrackingId:4d82b29d-b922-4634-a2ba-29fd14c40764_G16, SystemTracker:client-link18456132, Timestamp:2020-07-20T22:00:39, errorContext[NAMESPACE: eastus2-int-testing-ehns-punuljxb.servicebus.windows.net, PATH: eh0-int-stream, REFERENCE_ID: ccec2d406a3a451a85e4ffafab3f8d9a_G16, LINK_CREDIT: 296]
com.azure.core.amqp.exception.AmqpException: Idle link tracker, link eh0-int-stream has been idle for 1800000ms TrackingId:4d82b29d-b922-4634-a2ba-29fd14c40764_G16, SystemTracker:client-link18456132, Timestamp:2020-07-20T22:00:39, errorContext[NAMESPACE: eastus2-int-testing-ehns-punuljxb.servicebus.windows.net, PATH: eh0-int-stream, REFERENCE_ID: ccec2d406a3a451a85e4ffafab3f8d9a_G16, LINK_CREDIT: 296]
	at c.a.c.a.i.ExceptionUtil.toException(ExceptionUtil.java:78)
	at c.a.c.a.i.handler.LinkHandler.processOnClose(LinkHandler.java:90)
	at c.a.c.a.i.handler.LinkHandler.handleRemoteLinkClosed(LinkHandler.java:108)
	at c.a.c.a.i.handler.LinkHandler.onLinkRemoteClose(LinkHandler.java:50)
	at c.a.c.a.i.handler.SendLinkHandler.onLinkRemoteClose(SendLinkHandler.java:20)
	at org.apache.qpid.proton.engine.BaseHandler.handle(BaseHandler.java:176)
	at org.apache.qpid.proton.engine.impl.EventImpl.dispatch(EventImpl.java:108)
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.dispatch(ReactorImpl.java:324)
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.process(ReactorImpl.java:291)
	at c.a.c.a.i.ReactorExecutor.run(ReactorExecutor.java:82)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
INFO	2020-07-20 22:00:40,703	1806069	c.a.c.a.i.ReactorSession	[single-1]	linkName[eh0-int-stream]: Error occurred. Removing and disposing send link.
com.azure.core.amqp.exception.AmqpException: Idle link tracker, link eh0-int-stream has been idle for 1800000ms TrackingId:4d82b29d-b922-4634-a2ba-29fd14c40764_G16, SystemTracker:client-link18456132, Timestamp:2020-07-20T22:00:39, errorContext[NAMESPACE: eastus2-int-testing-ehns-punuljxb.servicebus.windows.net, PATH: eh0-int-stream, REFERENCE_ID: ccec2d406a3a451a85e4ffafab3f8d9a_G16, LINK_CREDIT: 296]
	at c.a.c.a.i.ExceptionUtil.toException(ExceptionUtil.java:78)
	at c.a.c.a.i.handler.LinkHandler.processOnClose(LinkHandler.java:90)
```

**Comment by Petermarcu** [comment]
Thanks!

**Comment by joshfree** [comment]
@conniey is this tracking for the August code complete?

**Comment by conniey** [comment]
Hey Josh, I am hoping to get this in for August code complete next week.

**Comment by conniey** [comment]
An update on this issue so it doesn't fall off. I am working on this still and it is slated for September release.

**Comment by vinceve** [comment]
Hi, we are experiencing this bug also. I reported it to Microsoft support as well.

**Comment by 123Jun321** [comment]
Hi @conniey, I want to know when the new version will be released? could you give a specific date?

**Comment by conniey** [comment]
Hey @123Jun321 This bug fix is slated for our September release, which will be the second week of September.

**Comment by 123Jun321** [comment]
HI, @conniey thanks! we will wait for it!

**Comment by roehrijn** [comment]
We have a similar issue here but I can't find any related exception in the logs. Only INFO messages and EventProcessorClient stops reading data after several days. We're going to wait for the September release and check again.

**Comment by 123Jun321** [comment]
Hi  @conniey  this is the second week of September, how is it going?

**Comment by fshimamoto** [comment]
Hi @conniey, we are facing the same issue here, is the library already updated?

**Comment by gsanchez-cit** [comment]
Hi @conniey, we too are facing this issue here, is the library already updated?

**Comment by conniey** [comment]
Hey, I have a PR open, #14990 that should resolve this. I am testing this PR before merging it for our release.

**Comment by conniey** [comment]
Should be resolved by #14990
